### PR TITLE
Chg 230968 align payment generator models between mit components and sps endpoint schema

### PR DESCRIPTION
**Pull request by yinkacocky**

### CODE REVIEW

File: EST.MIT.Payment.Models/PaymentRequest.cs [hunk 1]
```diff
@@ -6,25 +6,37 @@ public class PaymentRequest
...
-    [JsonProperty("deliveryBody")]
-    public string DeliveryBody { get; init; } = default!;
+    [JsonProperty("currency")]
+    public string Currency { get; init; } = default!;   
+    [JsonProperty("description")]
+    public string Description { get; init; } = default!;
+    [JsonProperty("originalInvoiceNumber")]
+    public string OriginalInvoiceNumber { get; init; } = default!;  
+    [JsonProperty("recoveryDate")]
+    public DateTime RecoveryDate { get; init; } = default!;
```
Comment (andygibbonsdefra) [inline EST.MIT.Payment.Models/PaymentRequest.cs:25]: Think OriginalSettlementDate (DateTime) is missing?

File: EST.MIT.Payment.Function/Validation/ValidateRequest.cs [hunk 2]
```diff
@@ -168,14 +157,17 @@ public static class ValidateRequest
...
+            ""paymentRequestsBatches""
         ]
         }";
 
     public static bool IsValid(string importRequest)
     {
         var schema = JSchema.Parse(schemaJson);
         var parseImportRequest = JObject.Parse(importRequest);
+
+        parseImportRequest.IsValid(schema);
```
Comment (andygibbonsdefra) [inline EST.MIT.Payment.Function/Validation/ValidateRequest.cs:169]: This looks a bit odd. I think the parseImportRequest that doesn't get returned can be removed?

File: EST.MIT.Payment.Models/PaymentRequest.cs (hunk 1 above)
Comment (yinkacocky) [reply EST.MIT.Payment.Models/PaymentRequest.cs]: Resolved

File: EST.MIT.Payment.Function/Validation/ValidateRequest.cs (hunk 2 above)
Comment (yinkacocky) [reply EST.MIT.Payment.Function/Validation/ValidateRequest.cs]: Resolved

File: EST.MIT.Payment.Models/PaymentRequest.cs [hunk 3]
```diff
@@ -6,25 +6,39 @@ public class PaymentRequest
...
-    [JsonProperty("deliveryBody")]
-    public string DeliveryBody { get; init; } = default!;
+    [JsonProperty("currency")]
+    public string Currency { get; init; } = default!;   
+    [JsonProperty("description")]
+    public string Description { get; init; } = default!;
+    [JsonProperty("originalInvoiceNumber")]
+    public string OriginalInvoiceNumber { get; init; } = default!;  
+    [JsonProperty("originalSettlementDate")]
+    public string OriginalSettlementDate { get; init; } = default!;
```
Comment (andygibbonsdefra) [inline EST.MIT.Payment.Models/PaymentRequest.cs:25]: OriginalSettlementDate amd RecoveryDate should be DateTime. (I thought RecoveryDate was DateTime just now?)

File: EST.MIT.Payment.Function/Validation/ValidateRequest.cs [hunk 4]
```diff
@@ -168,14 +178,14 @@ public static class ValidateRequest
...
-            ""invoices""
+            ""paymentRequestsBatches""
         ]
         }";
 
     public static bool IsValid(string importRequest)
     {
         var schema = JSchema.Parse(schemaJson);
-        var parseImportRequest = JObject.Parse(importRequest);
+        var parseImportRequest = JObject.Parse(importRequest);
```
Comment (andygibbonsdefra) [inline EST.MIT.Payment.Function/Validation/ValidateRequest.cs:188]: Being picky, I think there's a trailing space here now that is causing a 'diff' on this line to show. Be careful not to introduce white space!

**Review by andygibbonsdefra — CHANGES_REQUESTED** [review]

File: EST.MIT.Payment.Function/Validation/ValidateRequest.cs (hunk 4 above)
Comment (yinkacocky) [reply EST.MIT.Payment.Function/Validation/ValidateRequest.cs]: resolved

File: EST.MIT.Payment.Models/PaymentRequest.cs (hunk 3 above)
Comment (yinkacocky) [reply EST.MIT.Payment.Models/PaymentRequest.cs]: resolved

**Review by jbarnsley10 — APPROVED** [review]

**Review by andygibbonsdefra — APPROVED** [review]

**Pull request merged**